test/datasets/test_dataset_checkpoint.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+from torchtitan.checkpoint import DataLoaderWrapper
+from torchtitan.datasets.hf_datasets import build_hf_data_loader
+from torchtitan.datasets.tokenizer import create_tokenizer
+
+
+class TestDatasetCheckpoint:
```

Comment:
This test logic looks great! IIUC this is not in CI yet and needs to be run manually?

Also I think we need some sort of integration test as well -- basically save checkpoint at step T, and resume from checkpoint and assert we are loading normally at step T+1. In fact we are working on setting up examples to make it easier to add such a test. Let me check when can it land.

Reply:
It is in the CI. In cpu unit test yaml file, I see pytest command that runs at the top level. It should pick up the test. When I ran it locally, here is what I see:

>test/datasets/test_dataset_checkpoint.py::TestDatasetCheckpoint::test_c4_resumption PASSED                                                                                                               [ 14%]
test/test_job_config.py::TestJobConfig::test_command_line_args PASSED                                                                                                                                    [ 28%]
test/test_job_config.py::TestJobConfig::test_job_config_file PASSED                                                                                                                                      [ 42%]
test/test_job_config.py::TestJobConfig::test_job_file_does_not_exist PASSED                                                                                                                              [ 57%]
test/test_job_config.py::TestJobConfig::test_empty_config_file PASSED                                                                                                                                    [ 71%]
test/test_job_config.py::TestJobConfig::test_job_config_file_cmd_overrides PASSED                                                                                                                        [ 85%]
test/test_job_config.py::TestJobConfig::test_print_help PASSED                                                                                                                                           [100%]

---------- coverage: platform linux, python 3.10.14-final-0 ----------
Coverage XML written to file coverage.xml

Reply:
> Also I think we need some sort of integration test as well -- basically save checkpoint at step T, and resume from checkpoint and assert we are loading normally at step T+1. 

This would be nice but it might be outside the scope of this PR. The reason being that it this also requires setting up the run to be deterministic etc.

>  In fact we are working on setting up examples to make it easier to add such a test. Let me check when can it land.

Once this is available, we can add an integration test using that